### DISCUSSION TITLE
Any ideas/examples on how to persist a subfolder with an unknown name?

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [ScoopInstaller/Extras#7091] [General]
Basically, anything within a Themes folder should be persisted, but still allowed to be updated if it's part of the original software package.
Any idea how to deal with that?

related to https://github.com/ScoopInstaller/Extras/issues/4919#issuecomment-953263355

**Comment by HUMORCE** [comment]
in this case, an install script is needed for replace new default themes from `Themes.original` to `Themes`.

**Comment by issaclin32** [comment]
Currently *Scoop* does not support wildcards for `persist`.
You can manually persist the items in `pre_install` and `pre_uninstall` if you want to do that.
For example, if the you want to persist a folder named `settings-(A RANDOM STRING)`, then :
```
"pre_install": "if (Test-Path \"$persist_dir\\settings-*\") { Copy-Item \"$persist_dir\\settings-*\" \"$dir\\\" -Recurse }",
"pre_uninstall": [
    "if (Test-Path \"$dir\\settings-*\") {",
    "    ensure \"$persist_dir\"",
    "    Copy-Item \"$dir\\settings-*\" \"$persist_dir\\\"",
    "}"
],
```

**Comment by AkariiinMKII** [comment]
Scoop uses `New-Item` command to persist data, while `-Type HardLink` for files and `-Type Junction` for folders, scripts in `installer` or `pre_install` field would take the same effect. For unknown file names, using `Get-ChildItem` in pipeline may help.
But for the situation you mentioned, just move the folder extracted from package to persist directory, overwrite the conflicted files, and persist the folder.
